src/content-routing/types.d.ts

```diff
@@ -0,0 +1,11 @@
+export = ContentRouting;
+
+import PeerId from 'peer-id'
+import Multiaddr from 'multiaddr'
+import CID from 'cids'
+
+declare class ContentRouting {
+  constructor (options: Object);
+  provide (cid: CID): Promise<void>;
+  findProviders (cid: CID, options: Object): AsyncIterable<{ id: PeerId, multiaddrs: Multiaddr[] }>;
```

Comment:
maybe better to define a `Provider` here rather than an anonymous type for `{id, multiaddrs}`?

Reply:
This is more a representation of a peer rather then a Provider. As you pointed below, findPeer also has the same. We have been discussing about potentially change the API return type of this in https://github.com/libp2p/js-libp2p-interfaces/issues/85

We will probably look back into this